Makefile contain bash-ism

**Describe the bug**
The `build/targets.device.n0110.mak` makefile contain the `if [[ ... ]]` bash-sim without explicitly setting `SHELL=/bin/bash`.

**To Reproduce**
Steps to reproduce the behavior:
1. `make V=1 epsilon_flash`
2. Outputs:
```
....
if [[ "Bus 1 Device 013: ID 0483:df11" == *"0483:df11"* ]]; \
  then \
    python3 build/device/dfu.py -u output/release/device/n0110/flasher.light.dfu; \
    sleep 2; \
fi
/bin/sh: 1: [[: not found
```

**Expected behavior**
If `make V=1 SHELL=/bin/sh omega_flash` then the error does not occur.

**Desktop (please complete the following information):**
 - Omega Version: f052ad7bf8fe1fb243c69f25f278b1c5d3dc3960
 - Ubuntu 20.04

note that this causes the flash to fail, since the dfuse program isn't run and make swallows the error:
```
~/build/Omega: sudo make V=1 epsilon_flash
    ____                                  
   / __ \____ ___  ___  ____ _____ _     
  / / / / __ `__ \/ _ \/ __ `/ __ `/ 
 / /_/ / / / / / /  __/ /_/ / /_/ /       
 \____/_/ /_/ /_/\___/\__, /\__,_/    
                     /____/               
          - Omega does what Epsilon't     

Building O1.20.3-E14.4.1
  Targetting Numworks 
    Building on Linux amd64

DFU     epsilon_flash
INFO    About to flash your device. Please plug your device to your computer
        using an USB cable and press at the same time the 6 key and the RESET
        button on the back of your device.
until python3 build/device/dfu.py -l | grep -E "0483:a291|0483:df11" > /dev/null 2>&1; do sleep 2;done
if [[ "Bus 1 Device 013: ID 0483:df11" == *"0483:df11"* ]]; \
  then \
    python3 build/device/dfu.py -u output/release/device/n0110/flasher.light.dfu; \
    sleep 2; \
fi
/bin/sh: 1: [[: not found
python3 build/device/dfu.py -u output/release/device/n0110/epsilon.dfu
File: output/release/device/n0110/epsilon.dfu
    b'DfuSe' v1, image size: 1067045, targets: 1
    b'Target' 0, alt setting: 0, name: "ST...", size: 1066760, elements: 2
      0, address: 0x90000000, size: 1057448
      1, address: 0x08000000, size: 9296
    usb: 0483:a291, device: 0x0000, dfu: 0x011a, b'UFD', 16, 0x82d771f9
Writing memory...
0x90000000 1057448 [                         ]   0% Traceback (most recent call last):
  File "build/device/dfu.py", line 598, in <module>
    main()
  File "build/device/dfu.py", line 589, in main
    write_elements(elements, args.mass_erase, progress=cli_progress)
  File "build/device/dfu.py", line 516, in write_elements
    write_memory(addr, data[:write_size], progress,
  File "build/device/dfu.py", line 229, in write_memory
    set_address(xfer_base+xfer_bytes)
  File "build/device/dfu.py", line 206, in set_address
    raise Exception("DFU: set address failed")
Exception: DFU: set address failed
make: *** [build/targets.device.n0110.mak:17: epsilon_flash] Error 1
```

--- a/build/targets.device.n0110.mak
+++ b/build/targets.device.n0110.mak
@@ -13,7 +13,7 @@ $(BUILD_DIR)/test.external_flash.write.$(EXE): $(BUILD_DIR)/quiz/src/test_ion_ex
 	@echo "        button on the back of your device."
 	$(Q) until $(PYTHON) build/device/dfu.py -l | grep -E "0483:a291|0483:df11" > /dev/null 2>&1; do sleep 2;done
 	$(eval DFU_SLAVE := $(shell $(PYTHON) build/device/dfu.py -l | grep -E "0483:a291|0483:df11"))
-	$(Q) if [[ "$(DFU_SLAVE)" == *"0483:df11"* ]]; \
+	$(Q) if expr "$(DFU_SLAVE)" : ".*0483:df11.*" > /dev/null; \
 	  then \
 	    $(PYTHON) build/device/dfu.py -u $(word 2,$^); \
 	    sleep 2; \

